src/basic/strv.c

```diff
@@ -721,7 +721,7 @@ int strv_make_nulstr(char * const *l, char **ret, size_t *ret_size) {
         }
 
         if (!m) {
-                m = new0(char, 1);
+                m = new0(char, 2);
```

Comment:
hmm, this does not look right. 

first of all, you forgot to increase n by 1. this change of yours is hence without real effect.

but I think the comment is simply wrong, not the code. i.e an empty nulstr (i.e. one without even a single string) should be encoded as single NUL byte. because a nulstr consisting of two nul byte should instead be read as a nulstr consisting of one string: an empty one.

Reply:
Well, the current interface isn't exactly ideal either, see the workarounds that quentin has to do in the journald regex pr. I would either always terminate with 2 nul terminators or return the actual nulstr size in ret_size so that callers don't have to do complicated logic to figure out the actual length of the nulstr including nul terminators

Reply:
yeah, the status quo ante was fucked. I prepped a rework now in #25360. ptal.

unlike i though earlier you patch wasn't really wrong, but I think we should fix this properly, see the PR